Leanplum-SDK/Classes/Managers/LPLogManager.m

```diff
@@ -0,0 +1,143 @@
+//  Leanplum-iOS-SDK
+//
+//  Created by Dejan Krstevski on 31.07.20.
+//  Copyright (c) 2020 Leanplum, Inc. All rights reserved.
+//
+
+#import "LPLogManager.h"
+#import "LPUtils.h"
+#import "LPConstants.h"
+#import "LPRequestFactory.h"
+#import "LPRequestSender.h"
+
+
+@interface LPLogManager()
++ (void)maybeSendLog:(NSString *)message;
+@end
+@implementation LPLogManager
+static LPLogLevel logLevel = Info;
+
++(void)setLogLevel:(LPLogLevel)level
+{
+    @synchronized (self) {
+        logLevel = level;
+    }
+}
+
++(LPLogLevel)logLevel
+{
+    @synchronized (self) {
+        return logLevel;
+    }
+}
+
++ (void)maybeSendLog:(NSString *)message {
+    if (![LPConstantsState sharedState].loggingEnabled) {
+        return;
+    }
+
+    NSMutableDictionary *threadDict = [[NSThread currentThread] threadDictionary];
+    BOOL isLogging = [[[[NSThread currentThread] threadDictionary]
+                       objectForKey:LP_IS_LOGGING] boolValue];
+
+    if (isLogging) {
+        return;
+    }
+
+    threadDict[LP_IS_LOGGING] = @YES;
+
+    @try {
+        LPRequest *request = [LPRequestFactory logWithParams:@{
+                                                      LP_PARAM_TYPE: LP_VALUE_SDK_LOG,
+                                                      LP_PARAM_MESSAGE: message
+                                                      }];
+                [[LPRequestSender sharedInstance] sendEventually:request sync:NO];
+    } @catch (NSException *exception) {
+        LPLog(LPError, @"Unable to send log: %@", exception);
+    } @finally {
+        [threadDict removeObjectForKey:LP_IS_LOGGING];
+    }
+}
+
++ (void)logInternalError:(NSException *)e
+{
+    [LPUtils handleException:e];
+    if ([e.name isEqualToString:@"Leanplum Error"]) {
+        @throw e;
+    }
+    
+    for (id symbol in [e callStackSymbols]) {
+        NSString *description = [symbol description];
+        if ([description rangeOfString:@"+[Leanplum trigger"].location != NSNotFound
+            || [description rangeOfString:@"+[Leanplum throw"].location != NSNotFound
+            || [description rangeOfString:@"-[LPVar trigger"].location != NSNotFound
+            || [description rangeOfString:@"+[Leanplum setApiHostName"].location != NSNotFound) {
+            @throw e;
+        }
+    }
+    NSString *versionName = [[[NSBundle mainBundle] infoDictionary]
+                             objectForKey:@"CFBundleVersion"];
+    if (!versionName) {
+        versionName = @"";
+    }
+    int userCodeBlocks = [[[[NSThread currentThread] threadDictionary]
+                           objectForKey:LP_USER_CODE_BLOCKS] intValue];
+    if (userCodeBlocks <= 0) {
+        @try {
+            LPRequest *request = [LPRequestFactory logWithParams:@{
+                                     LP_PARAM_TYPE: LP_VALUE_SDK_LOG,
+                                     LP_PARAM_MESSAGE: [e description],
+                                     LP_KEY_STACK_TRACE: [[e callStackSymbols] description] ?: @"",
```

Comment:
make sure we send stacktrace as part of the message since api doesnt accept stacktrace param